pkg/registry/core/pod/storage/eviction.go

```diff
@@ -303,6 +304,17 @@ func canIgnorePDB(pod *api.Pod) bool {
 	return false
 }
 
+// enoughHealthyReplicas checks to determine there are enough healthy replicas
+// available to allow already disrupted pods to be evicted.  We also ensure the
+// PDB is not in failSafe mode so we don't use stale data.
+func enoughHealthyReplicas(pdb *policyv1.PodDisruptionBudget) bool {
+	// We only consider if the conditions is not nil and if we're in failSafe mode represented by
+	// condition.Reason == policyv1.SyncFailedReason
+	condition := apimeta.FindStatusCondition(pdb.Status.Conditions, policyv1.DisruptionAllowedCondition)
+	return condition != nil && !(condition.Status == metav1.ConditionFalse && condition.Reason == policyv1.SyncFailedReason) &&
```

Comment:
nit that would really help me read this

```go
// we only allow disruptions for pods that have synced at least once
// TODO deads2k wonders why we would block pod deletes if the PDB hasn't been able to be evaluated at least once
if condition == nil{
  return  false
}
// if the sync failed, then we cannot disrupt this pod because of a very important reason
if condition.Status == false && reason == syncfailed{
  return false
}


return pdb.Status.CurrentHealthy >= pdb.Status.DesiredHealthy && pdb.Status.DesiredHealthy > 0
```

Reply:
it's intended to be programmatically readable (as opposed to message, which is human-readable), and there's a fair amount of prior art:

https://github.com/kubernetes/kubernetes/blob/56efa75173922e93af6ef146f50422a22680cc46/pkg/controller/deployment/util/deployment_util.go#L775-L798

https://github.com/kubernetes/kubernetes/blob/56efa75173922e93af6ef146f50422a22680cc46/pkg/controller/deployment/progress.go#L48-L49

https://github.com/kubernetes/kubernetes/blob/56efa75173922e93af6ef146f50422a22680cc46/pkg/controller/util/node/controller_utils.go#L103-L108

https://github.com/kubernetes/kubernetes/blob/56efa75173922e93af6ef146f50422a22680cc46/staging/src/k8s.io/apiextensions-apiserver/pkg/controller/nonstructuralschema/nonstructuralschema_controller.go#L111-L113

https://github.com/kubernetes/kubernetes/blob/56efa75173922e93af6ef146f50422a22680cc46/staging/src/k8s.io/apiextensions-apiserver/pkg/controller/nonstructuralschema/nonstructuralschema_controller.go#L125-L127